src/library/vm/vm_pos_info_provider.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Author: Jared Roesch
+*/
+#include "library/vm/vm_pos_info.h"
+#include "kernel/pos_info_provider.h"
+#include "library/vm/vm_nat.h"
+#include "library/vm/vm_expr.h"
+#include "library/vm/vm_option.h"
+
+namespace lean {
+
+struct vm_pos_info_provider : public vm_external {
+    pos_info_provider const & m_val;
```

Comment:
Yeah I agree. I was worried about this yesterday, but no one was awake/around to ask about the right way to do this. I'll refactor it to use this approach instead.